src/base/static/components/organisms/site-header.js

```diff
@@ -0,0 +1,398 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import styled from "react-emotion";
+import { connect } from "react-redux";
+
+import { SiteLogo } from "../atoms/imagery";
+import { Button } from "../atoms/buttons";
+import { Link } from "../atoms/navigation";
+import UserMenu from "../molecules/user-menu";
+import { RegularTitle } from "../atoms/typography";
+
+import { navBarConfigSelector } from "../../state/ducks/nav-bar-config";
+import { appConfigSelector } from "../../state/ducks/app-config";
+import { currentTemplateSelector } from "../../state/ducks/ui";
+import {
+  isLeftSidebarExpandedSelector,
+  setLeftSidebarComponent,
+  setLeftSidebarExpanded,
+} from "../../state/ducks/left-sidebar";
+import mq from "../../../../media-queries";
+
+// TODO: Make the outermost div a header element when we dissolve base.hbs.
+// Right now the header element lives in base.hbs.
+const SiteHeaderWrapper = styled("div")(props => ({
+  backgroundColor: props.theme.bg.default,
+  display: "flex",
+  height: "100%",
+  alignItems: "center",
+
+  [mq[0]]: {
+    flexDirection: "column",
+    padding: "8px",
+  },
+  [mq[1]]: {
+    flexDirection: "row",
+  },
+}));
+
+const NavContainer = styled("nav")(props => ({
+  [mq[0]]: {
+    display: props.isHeaderExpanded ? "flex" : "none",
```

Comment:
I used this trick a few times in this component: using state to trigger a `display: none` on mobile layouts. I was trying to keep all logic for desktop/mobile differences in the CSS via media queries (as opposed to introducing a way of tracking this in the JS).